x/proof/keeper/msg_server_submit_proof.go

```diff
@@ -191,3 +201,110 @@ func (k Keeper) deductProofSubmissionFee(ctx context.Context, supplierOperatorAd
 
 	return nil
 }
+
+// proofRequirementForClaim checks if a proof is required for a claim.
+// If it is not, the claim will be settled without a proof.
+// If it is, the claim will only be settled if a valid proof is available.
+// TODO_BLOCKER(@bryanchriswhite, #419): Document safety assumptions of the probabilistic proofs mechanism.
+func (k Keeper) ProofRequirementForClaim(ctx context.Context, claim *types.Claim) (_ types.ProofRequirementReason, err error) {
```

Comment:
Did you make changes here or just copy-paste? Haven't re-reviewed it yet.

Reply:
Just copy pasting, imports got adjusted though.